src/MSBuild/XMake.cs

```diff
@@ -1725,7 +1725,7 @@ internal static void GatherCommandLineSwitches(ArrayList commandLineArgs, Comman
                             if (String.Equals(switchName, "m", StringComparison.OrdinalIgnoreCase) ||
                                 String.Equals(switchName, "maxcpucount", StringComparison.OrdinalIgnoreCase))
                             {
-                                int numberOfCpus = Environment.ProcessorCount;
+                                int numberOfCpus = NativeMethodsShared.GetLogicalCoreCount();
```

Comment:
```suggestion
                                int numberOfCpus = Environment.ProcessorCount;
```

The change wave won't apply otherwise. Which actually suggests that we should consider putting all of the logic here into NativeMethodsShared since it can be centralized.

Reply:
I moved the xmake logic instead: https://github.com/dotnet/msbuild/pull/6014/commits/d5ac55a50293585ce4905726737a2efb82fbb0b4
I think this is the correct way to do it, didn't realize the thing can return -1 :)